File: src/main/java/codereview/simpleorder/repository/command/ItemRepository.java
package codereview.simpleorder.repository.command;

import codereview.simpleorder.domain.item.Item;
import jakarta.persistence.LockModeType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Lock;

import java.util.List;

public interface ItemRepository extends JpaRepository<Item, Long> {

    @Lock(value = LockModeType.OPTIMISTIC)
    List<Item> findByIdIn(List<Long> itemIds);
}

File: item/src/main/kotlin/io/philo/shop/repository/ItemRepository.kt
package io.philo.shop.repository

import io.philo.shop.domain.entity.ItemEntity
import org.springframework.data.jpa.repository.JpaRepository

interface ItemRepository : JpaRepository<ItemEntity, Long> {
    
    fun findByIdIn(itemIds: Collection<Long>): List<ItemEntity>
}